packages/purpose-process/src/services/purposeService.ts

```diff
@@ -575,6 +589,63 @@ export function purposeServiceBuilder(
         totalCount: purposesList.totalCount,
       };
     },
+    async createPurposeVersion({
+      purposeId,
+      seed,
+      organizationId,
+      correlationId,
+      logger,
+    }: {
+      purposeId: PurposeId;
+      seed: ApiPurposeVersionSeed;
+      organizationId: TenantId;
+      correlationId: string;
+      logger: Logger;
+    }): Promise<PurposeVersion> {
+      logger.info(`Creating Version for Purpose ${purposeId}`);
+
+      const purpose = await retrievePurpose(purposeId, readModelService);
+      assertOrganizationIsAConsumer(organizationId, purpose.data.consumerId);
+      assertDailyCallsIsDifferentThanBefore(purpose.data, seed.dailyCalls);
+
+      const eservice = await retrieveEService(
+        purpose.data.eserviceId,
+        readModelService
+      );
+
+      const ownership = getOrganizationRole({
+        organizationId,
+        producerId: eservice.producerId,
+        consumerId: purpose.data.consumerId,
+      });
```

Comment:
Isn't this redundant after `assertOrganizationIsAConsumer`?

Reply:
No, the caller may still be consumer or both consumer and producer